infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/receive/LogEntryWriter.java

```diff
@@ -175,6 +175,9 @@ public boolean apply(LogReplicationEntryMsg msg) throws ReplicationWriterExcepti
      * @param ackTimestamp
      */
     public void reset(long snapshot, long ackTimestamp) {
+        // Sync with registry table when LogEntryWriter is reset, which will happen when Snapshot sync is completed, and
+        // when LogReplicationSinkManager is initialized and reset.
+        config.syncWithRegistry();
```

Comment:
the sync is required after each ACK.  reset() is invoked only on leadership change and role change.